bin/postgresql-upgrade.in

```diff
@@ -163,6 +163,27 @@ if [ ! -x "$upgradefrom_engine/postgres" ]; then
     exit 1
 fi
 
+# Check for data_directory entry in config file
+# valid entry means that custom PGDATA path is present which is not supported
+# BZ (#1935301)
+conffile="$option_datadir/postgresql.conf"
+
+test -r "$conffile" || {
+    error "config file $conffile is not readable or does not exist"
+    die "Old cluster in '$data' does not seem to be initialized"
+}
+
+sp='[[:space:]]'
+sed_expr_pgdata="s/^$sp*data_directory$sp*=$sp[\'|\"]\(.*\+\)[\'|\"].*/\1/p"
+conf_pgdata=`sed -n "$sed_expr_pgdata" $conffile | tail -1`
+
+test -n "$conf_pgdata" && {
+    error   $"data_directory field in configuration file ($conffile)"
+    error_q $"is not supported. db datadir (PGDATA) cannot be specified"
+    error_q $"in configuration file."
```

Comment:
What can be also added is some recommendation, like: "For using this script for upgrade, make sure the data are located in the directory that matches the PGDATA variable inside the service/unit file."

Reply:
Yes, I added additional information for both scripts.